tests/sles4sap/redirection_tests/ensa2_setup_Netweaver_WebGUI_access.pm

```diff
@@ -0,0 +1,156 @@
+
+package ensa2_setup_Netweaver_WebGUI_access;
+use testapi;
+use sles4sap::console_redirection;
+use sles4sap::console_redirection::redirection_data_tools;
+use sles4sap::sap_host_agent qw(saphostctrl_list_instances);
+use sles4sap::sapcontrol qw(sapcontrol);
+
+=head1 NAME
+
+sles4sap/redirection_tests/ensa2_setup_Netweaver_WebGUI_access.pm.pm - Setup Netweaver WebGUI access on PAS for netweaver instances.
+
+=head1 MAINTAINER
+
+QE-SAP <qe-sap@suse.de>
+
+=head1 DESCRIPTION
+
+Module setup Netweaver WebGUI access for netweaver instances.
+
+B<The key tasks performed by this module include:>
+
+=over
+
+=item * Setup ICM http port on PAS instance via cmdline
+
+=item * Restart NW instance
+
+=item * Configure WebGUI services (via SQL statement)
+
+=item * Activate WebGUI services (via SQL statement)
+
+=item * Restart NW instance again
+
+=item * Check URL is accessible on NW hosts
+
+=back
+
+=head1 OPENQA SETTINGS
+
+Test module does not use any openQA settings
+
+=cut
+
+sub run {
+    my ($self, $run_args) = @_;
+    my $redirection_data = sles4sap::console_redirection::redirection_data_tools->new($run_args->{redirection_data});
+
+    # Get PAS host
+    my %pas_host = %{$redirection_data->get_pas_host};
+    if (!%pas_host) {
+        record_info('N/A', 'Netweaver deployment not detected, skipping.');
+        return;
+    }
+
+    # Everything is now executed on SUT, not worker VM
+    my $host = join('', keys(%pas_host));
+    my $ip_addr = $pas_host{$host}{ip_address};
+    my $user = $pas_host{$host}{ssh_user};
+    die "Redirection data missing. Got:\nIP: $ip_addr\nUSER: $user\n" unless $ip_addr and $user;
+    record_info("Host: $host");
+    connect_target_to_serial(destination_ip => $ip_addr, ssh_user => $user, switch_root => 'yes');
+
+    # Setup ICM http port via cmdline
+    my $sid = get_required_var('SAP_SID');
+    my $host_fqdn = script_output('hostname -f');
+    my $profile_dir = "/sapmnt/$sid/profile/";
+    my $profile = "$profile_dir" . script_output("ls $profile_dir | grep $host | grep -v '\\.'");
+    record_info("profile: $profile");
+    assert_script_run("cp $profile $profile.bk");
+    assert_script_run("echo 'icm/server_port_0 = PROT=HTTP,PORT=8080' >> $profile");
+    assert_script_run("echo icm/host_name_full = $host_fqdn >> $profile");
+
+    # Restart NW instance
+    my $instance_data = saphostctrl_list_instances(as_root => 'yes', running => 'yes');
+    my $output = sapcontrol(
+        webmethod => 'RestartInstance',
+        instance_id => $instance_data->[0]{instance_id},
+        sidadm => $instance_data->[0]{sap_sid},
+        return_output => 'yes');
+
+    # Configure WebGUI services (via SQL statement)
+    # Userstore key check
+    assert_script_run('su - qesadm -c "hdbuserstore list"');
+    assert_script_run('su - qesadm -c "hdbsql -U DEFAULT \'select * from m_services\'" | tee');
+
+    # Activate WebGUI services (via SQL statement)
+    my @hdbsql = ();
```

Comment:
```
my $sql_base = 'hdbsql -U DEFAULT update ICFSERVLOC SET ICFACTIVE = \\'X\\' where ICF_NAME =';
push(@hdbsql, "$sql_base \\'MIMES\\' and ICFPARGUID = \\'0CMVCTDU211SGUZWM853S6A3S\\'");
```

Or define a list of couple `name/UID` and move the push in a foreach

Reply:
Done. Revised code according to Alvaro's comments.